### ISSUE TITLE
A question about electron systematics

### ISSUE THREAD

**Issue by MihaMuskinja** [UCATLAS/xAODAnaHelpers#703]
Dear developers,

I was exploring how the electron systematics are handled in the framework and I stumbled upon some problems. Please find my questions below.

**How to use the "TOTAL correlation" model of electron systematics, explained here: https://twiki.cern.ch/twiki/bin/view/AtlasProtected/ElectronEfficiencyCorrelationModel ?**

> TOTAL: Only 1 parameter named "TOTAL_1NPCOR_PLUS_UNCOR"

the advantage of this model is that there is only one systematic for trigger/iso/reco/ID instead of several. The problem is that m_asgElEffCorrTool_elSF_PID->recommendedSystematics() (https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/ElectronEfficiencyCorrector.cxx#L218) gets only the separate systematics (NP0,NP1, ... ,NP14) and not the TOTAL_1NPCOR_PLUS_UNCOR. This is also seen in the debug stream:

> HelperFunctions::getLi... INFO    systName EL_EFF_ID_ID_CorrUncertaintyNP0__1up
> HelperFunctions::getLi... INFO      EL_EFF_ID_CorrUncertaintyNP0__1down
> HelperFunctions::getLi... INFO      EL_EFF_ID_CorrUncertaintyNP0__1up
> HelperFunctions::getLi... INFO      EL_EFF_ID_CorrUncertaintyNP10__1down
> HelperFunctions::getLi... INFO      EL_EFF_ID_CorrUncertaintyNP10__1up
> HelperFunctions::getLi... INFO      EL_EFF_ID_CorrUncertaintyNP11__1down
> ...
> ...

These systematics are an overkill for our analysis, as it is sufficient to use only the TOTAL instead of all for us.

**How to select several electron systematics and how to write the nominal SF in addition to the up/down SFs?**

So far the only command that worked for me was:

"m_systNamePID"           : "EL_EFF_ID_CorrUncertaintyNP0",

I tried adding several systematics to the string: "EL_EFF_ID_CorrUncertaintyNP0,EL_EFF_ID_CorrUncertaintyNP1", but that didn't work.

An additional issue is that the nominal SF doesn't seem to be written to the output file with the "m_systNamePID"           : "EL_EFF_ID_CorrUncertaintyNP0", option. I only get 2 entries in the SF std::vector< float > per electron, as shown below:

```
***********************************************************
*    Row   * Instance * el_pt/1e3 *    el_eta * el_PIDEff *
***********************************************************
*        9 *        0 * 213.92128 * 0.7739586 * 0.9984400 *
*        9 *        1 * 141.72603 * 0.1268026 * 0.9987599 *
*        9 *        2 *           *           * 0.9992200 *
*        9 *        3 *           *           * 0.9992600 *

```

Here I print out the event number (Row), electron number (Instance), electron pt, electron eta, and electron ID scale factor vector. We can see that the electron with pt=213 and eta=0.77 has two scale-factors: 0.9984400 and 0.9987599. By looking at the actual scale-factor root file I noticed that the nominal scale factor for this pt/eta is 0.9986 and corr0 sys is 0.0086:

0.9986 + 0.0086 = 0.99876 (0.9987599)
0.9986 - 0.0086 = 0.99844 (0.9984400)

This implies that the nominal SF is not written and only 1up/1down SFs are written. The same is true for the second electron: 0.99924 +- 0.00002 .

Best,
Miha

**Comment by MihaMuskinja** [comment]
Maybe we could start with a simple issue:

How to include several systematic uncertainties for electrons at once? I'm talking about the configuration of the ElectronEfficiencyCorrector tool via the m_systNameXXX members.

**Comment by kratsg** [comment]
@MihaMuskinja not sure if solved, but it appears to be as #808 is closed.
